django/db/models/expressions.py

```diff
@@ -873,12 +879,13 @@ def __repr__(self):
 
 class When(Expression):
     template = 'WHEN %(condition)s THEN %(result)s'
+    conditional = False  # May not be used within a WHERE without a wrapping Case().
```

Comment:
This seems a bit of an abuse of _this expression cannot be used by itself_ (e.g. it should be disallowed in `.annotate`) as well but that should do for now.